repository.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+)
+
+// findRepository looks for a repository directory starting from the
+// given directory, going up one level until the root directory or
+// until the repository is found.  Returns os.NotExists in case the
+// algorithm ran successfully but found no repository.
```

Comment:
And thank you for this lesson. It is good lesson about testing in go this below. Thank you so much!!

Reply:
np this was fun :)